src/components/ClauseComponent.js

```diff
@@ -190,20 +196,65 @@ function ClauseComponent(props) {
       </S.ClauseHeader>
       { !props.readOnly
         && <>
-          <S.TestWrapper {...iconWrapperProps}>
-            <S.ClauseIcon {...testIconProps}>
+          <S.TestWrapper 
+            {...iconWrapperProps}
+            onMouseEnter={() => setHoveringTestIcon(true)}
+            onMouseLeave={() => setHoveringTestIcon(false)}
+            onClick={testIconProps.onClick}
```

Comment:
I think `onClick` here should be `() => clauseProps.CLAUSE_TEST_FUNCTION(props)`, and then that function should be removed from `testIconProps`

Reply:
Okay will do!